Improve and extend changelog logs (#74106)
ansible/ansible: docs/docsite/rst/community/development_process.rst

Changelogs
----------

> Changelogs help users and developers keep up with changes to Ansible and Ansible collections. Ansible and many collections build changelogs for each release from fragments. For ansible-core and collections using this model, you **must** add a changelog fragment to any PR that changes functionality or fixes a bug. You do not have to add a changelog fragment for PRs that add new modules and plugins, because our tooling does that for you automatically.

Reviewer: ```suggestion
Changelogs help users and developers keep up with changes to ansible-core and Ansible collections. Ansible and many collections build changelogs for each release from fragments. For ansible-core and collections using this model, you **must** add a changelog fragment to any PR that changes functionality or fixes a bug.

You do not need a changelog fragment for PRs that:

* add new modules and plugins, because our tooling does that for you automatically;
* contain only documentation changes;

.. note::
  Some collections require a changelog fragment for every pull request. They use the ``trivial:`` section for entries mentioned above that will be skipped when building a release changelog.

```
Should we get rid of the period before `because`?

Also `Changelogs help users and developers keep up with changes to Ansible and Ansible collections.`
I'd change this to:
```
Changelogs help users and developers keep up with changes to Ansible.
```
because Ansible collections are part of Ansible
or, if ansible-core is implied, to
```
Changelogs help users and developers keep up with changes to Ansible-core and Ansible collections.
```
and in the next sentence too
(not sure if it should be Ansible-core, Ansible-Core, or ansible-core)

Also FYI: I'll replace all PRs to "pull requests" in another PR later, it's too many places to fix now.

Reviewer: @felixfontein your thoughts ^

Author: Can you adjust your suggestion so that it uses `;`/`.` for the list? Then I'd merge it.

About the second part, I'd rather use `Changelogs help users and developers keep up with changes to ansible-core and Ansible collections.` (the correct spelling is `ansible-core` with lower-case `a`). Also not all collections are part of Ansible :)

Reviewer: Done, PTAL. If it's not what you meant, feel free to change it yourself, np

Reviewer: After that, could you please rephrase the second and the third sentences: 1) it starts with Ansible and 2) there are, IMO, too many repetitions in the first paragraph

More precisely:
* Every bugfix PR must have a changelog fragment. The only exception are fixes to a change that has not yet been included in a release.
* Every feature PR must have a changelog fragment.
* New modules and plugins (except jinja2 filter and test plugins) must have ``versions_added`` set correctly, and do not need a changelog fragment. The tooling detects new modules and plugins by their ``versions_added`` value and announces them in the next release's changelog automatically.

> * New jinja2 filter and test plugins, and also new roles and playbooks (for collections) must have a changelog fragment. See :ref:`changelogs_how_to_format_j2_roles_playbooks` or the `antsibull-changelog documentation for such changelog fragments <https://github.com/ansible-community/antsibull-changelog/blob/main/docs/changelogs.rst#adding-new-roles-playbooks-test-and-filter-plugins>_` for information on how the fragments should look like.

Reviewer 2: @felixfontein FYI https://jakubmarian.com/how-vs-what-does-it-look-like-in-english/ / https://new.testmagic.com/sat-questions/what-vs-how/

Changelogs
----------

> Changelogs help users and developers keep up with changes to ansible-core and Ansible collections. Ansible and many collections build changelogs for each release from fragments. For ansible-core and collections using this model, you **must** add a changelog fragment to any PR that changes functionality or fixes a bug.

Author: @Andersson007 I don't think it's a good idea to remove any use of Ansible/ansible here, as it will destroy the meaning. Maybe ok would be:
```suggestion
Changelogs help users and developers keep up with changes to ansible-core and Ansible collections. Ansible and many collections build changelogs for each release from fragments. For them, you **must** add a changelog fragment to any PR that changes functionality or fixes a bug.
```
but even that doesn't sound right to me.

Reviewer: we still have `.. Ansible and many collections..`. If we're talking about ansible-core it should be `ansible-core` but if we start with it, it should start with a capital letter:) Can we rephrase the sentence to put `ansible-core`, say, in the middle of the sentence?

You do not need a changelog fragment for PRs that:

> * add new modules and plugins, because our tooling does that for you automatically;

Reviewer 3: ```suggestion
* add new modules and plugins, because Ansible tooling does that automatically;
```

After you have written the changelog fragment for your PR, commit the file and include it with the pull request.

.. _changelogs_how_to_format_j2_roles_playbooks:

> Changelog fragment entry format for new jinja2 plugins, roles and playbooks

Reviewer: ```suggestion
Changelog fragment entry format for new jinja2 plugins, roles, and playbooks
```

> ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Author: ```suggestion
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

> While new modules and plugins that are not jinja2 filter or test plugins are mentioned automatically in the generated changelog, jinja2 filter and test plugins, roles and playbooks are not. To make sure they are mentioned, a changelog fragment in a specific format is needed:

Reviewer: ```suggestion
While new modules and plugins that are not jinja2 filter or test plugins are mentioned automatically in the generated changelog, jinja2 filter and test plugins, roles, and playbooks are not. To make sure they are mentioned, a changelog fragment in a specific format is needed:
```
